src/psyclone/gocean1p0.py

```diff
@@ -2385,11 +2388,15 @@ def psyir_expression(self):
                             " 'field' or 'scalar' but found '{0}'."
                             "".format(self.argument_type))
 
-    def infer_datatype(self):
+    def infer_datatype(self, symbol_table):
```

Comment:
You might as well replace `self._call` with `symbol_table` if do don't mind.

Reply:
I've removed the `symbol_table` argument.